test/IntegrationTests/MSTest.Acceptance.IntegrationTests/InconclusiveTests.cs

```diff
@@ -0,0 +1,235 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Testing.Platform.Acceptance.IntegrationTests;
+using Microsoft.Testing.Platform.Acceptance.IntegrationTests.Helpers;
+using Microsoft.Testing.Platform.Helpers;
+
+namespace MSTest.Acceptance.IntegrationTests;
+
+[TestClass]
+public sealed class InconclusiveTests : AcceptanceTestBase<InconclusiveTests.TestAssetFixture>
+{
+    public enum Lifecycle
+    {
+        AssemblyInitialize,
+        ClassInitialize,
+        TestInitialize,
+        TestMethod,
+        TestCleanup,
+        ClassCleanup,
+        AssemblyCleanup,
+    }
+
+    [TestMethod]
+    [DataRow(Lifecycle.AssemblyInitialize)]
+    [DataRow(Lifecycle.ClassInitialize)]
+    [DataRow(Lifecycle.TestInitialize)]
+    [DataRow(Lifecycle.TestMethod)]
+    [DataRow(Lifecycle.TestCleanup)]
+    [DataRow(Lifecycle.ClassCleanup)]
+    [DataRow(Lifecycle.AssemblyCleanup)]
+    public async Task TestOutcomeShouldBeRespectedCorrectly(Lifecycle inconclusiveStep)
```

Comment:
It'd be nice to have the "values source" attribute for such case

Reply:
I'll merge as we discussed.